lib/runtime/src/pipeline/network/egress/push_router.rs

```diff
@@ -179,6 +186,29 @@ where
         tracing::debug!("router generate");
         self.addressed.generate(request).await
     }
+
+    async fn generate_with_fault_tolerance<F, R>(
```

Comment:
Could we capture this in an AsyncEngine and then use it with just .generate?

Reply:
Not sure if I understand, this method is private to the PushRouter impl, and only called by PushRouter impl methods. Why are we moving this to AsyncEngine?